tt_metal/hw/inc/dataflow_api.h

```diff
@@ -1438,6 +1454,47 @@ void noc_semaphore_set(volatile tt_l1_ptr uint32_t* sem_addr, uint32_t val) {
     (*sem_addr) = val;
 }
 
+
+/**
+ * Initiates an asynchronous write of a 32-bit value to a NOC destination.
+ * Typically used for writing registers, but can be used for memory locations as well.
+ * The destination is specified as a 64-bit NOC address (see \a noc_async_write).
+ * The advantage over using \a noc_async_write is that we don't a Tensix L1
+ * memory source location; the write value is written directly into a register.
+ * Unlike using \a noc_async_write, there are also no address alignment concerns.
+ * Also, see \a noc_async_write_barrier.
+ *
+ * The destination node can be either a DRAM bank, Tensix core+L1 memory
+ * address or a PCIe controller.
+ *
+ * Return value: None
+ *
+ * | Argument  | Description                                                    | Type     | Valid Range                                                   | Required |
+ * |-----------|----------------------------------------------------------------|----------|---------------------------------------------------------------|----------|
+ * | addr      | Encoding of the destination location (x,y)+address             | uint64_t | DOX-TODO(insert a reference to what constitutes valid coords) | True     |
+ * | val       | The value to be written                                        | uint32_t | Any uint32_t value                                            | True     |
+ * | be        | Byte-enable                                                    | uint8_t  | 0x1-0xF                                                       | False    |
+ */
+FORCE_INLINE
+void noc_inline_dw_write(uint64_t addr, uint32_t val, uint8_t be = 0xF) {
+
+    DEBUG_STATUS('N', 'S', 'I', 'W');
```

Comment:
uniquify (SI is semaphore inc, this could be ID)

Reply:
Also fixed in the latest commit.